### PR TITLE
Serial and devDependencies

### PR DESCRIPTION
**Pull request by dtex**
See this commit https://github.com/jgautier/firmata/commit/7388429376472fd6c9a53c5d61e4d93a205a8e49

### CODE REVIEW

File: lib/constants.js [hunk 1]
```diff
@@ -46,6 +46,23 @@ module.exports.STEPPER = Object.freeze({
   }
 });
 
+module.exports.SERIAL_MODES = {
+  READ_CONTINUOUS: 0x00,
+  STOP_READING: 0x01
+};
+
+// ids for hardware and software serial ports on the board
+module.exports.SERIAL_PORT_IDs = {
```
Comment (soundanalogous) [inline lib/constants.js]: There are also the constants that map UART pins: https://github.com/jgautier/firmata/blob/master/lib/firmata.js#L482-L492

File: lib/board-io.js [hunk 2]
```diff
@@ -288,6 +292,71 @@ BoardIO.prototype.i2cReadOnce = function() {
 };
 
 /**
+ * Sends a config request to the board for hardware or sotware serial
+ * with a required opts parameter that must include the serial port ID,
+ * baud rate, and if software serial, the transmit and receive pins
+ * @param {number} delay in microseconds to set for I2C Read
```
Comment (soundanalogous) [inline lib/board-io.js]: The param documentation is incorrect. Use this instead: https://github.com/jgautier/firmata/blob/master/lib/firmata.js#L1487-L1492

Comment (rwaldron) [inline lib/board-io.js]: We need a "copy pasta" emoji ;)

File: lib/constants.js (hunk 1 above)
Comment (dtex) [inline lib/constants.js]: Added these

**Comment by dtex** [comment]
Thanks @soundanalogous for the catch on the serialRead comments. I've made those changes/additions.

**Comment by dtex** [comment]
:copyright:  :spaghetti: ?

**Comment by dtex** [comment]
Hi @achingbrain I just wanted to ping on this. I've got the new GPS class for Johnny-Five all ready to go but the unit tests won't work until the serial methods are available in our mocks.

Thanks

**Comment by achingbrain** [comment]
@dtex sorry for the delay - I'm actually on holiday at the moment with only sporadic internet access..

**Comment by achingbrain** [comment]
Published as 3.0.5

**Pull request merged**